src/libUtils/TimeUtils.h

```diff
@@ -18,12 +18,11 @@
 #ifndef __TIMEUTILS_H__
 #define __TIMEUTILS_H__
 
-#include <time.h>
-
+#include <chrono>
 #include <boost/multiprecision/cpp_int.hpp>
 
-struct timespec r_timer_start();
-double r_timer_end(struct timespec start_time);
+std::chrono::system_clock::time_point r_timer_start();
+double r_timer_end(std::chrono::system_clock::time_point start_time);
```

Comment:
💭 It may be desirable to return `std::microseconds` instead of just `double`, but I left that for a future change.